### PR TITLE
28-CampaignPageUI

### PR DESCRIPTION
**Pull request by TylerCarey98**
-Added UI to the campaign page
-Added backend management for the campaign page
-Changed spelling on things
-Added notifications to the campaign page

### CODE REVIEW

File: src/DnDTracker.Web/Views/Campaign/Create.cshtml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                                closeButton: true
+                            });
+                            notify.overlay(true);
+                        });
+                        break;
+                    case "err":
+                        console.error(json.message);
+
+                        // Use notify.js to give the user a "failure" notification
+                        $('#create-button').click(function () {
```
Comment (itsmistad) [inline src/DnDTracker.Web/Views/Campaign/Create.cshtml]: Remove this on click event handler. Calling this everytime you click create and successfully create a campaign will add _another_ on click event handler to the stack.

File: src/DnDTracker.Web/Views/Campaign/Create.cshtml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                    case "ok":
+                        console.log(json.message);
+
+                        // Clear the input fields
+                        $('#create-campaign__name').val("");
+                        $('#create-campaign__description').val("");
+                        $('#create-campaign__information').val("");
+
+                        // Use notify.js to give the user a "success" notification
+                        $('#create-button').click(function () {
```
Comment (itsmistad) [inline src/DnDTracker.Web/Views/Campaign/Create.cshtml]: Remove this on click event handler. Calling this everytime you click create and successfully create a campaign will add _another_ on click event handler to the stack.

This also breaks functionality on the first time and breaks the notify.me OK button

File: src/DnDTracker.Web/Views/Campaign/Create.cshtml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                            notify.me({
+                                class: centerPopupClasses,
+                                header: 'Campaign Created Successfully!',
+                                subheader: '',
+                                body: 'Your campaign will now show in the view campaigns page! Click below to return to the page.',
```
Comment (itsmistad) [inline src/DnDTracker.Web/Views/Campaign/Create.cshtml]: Check Slack for recommendations on these line items.

File: src/DnDTracker.Web/Views/Campaign/Create.cshtml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                    case "err":
+                        console.error(json.message);
+
+                        // Use notify.js to give the user a "failure" notification
+                        $('#create-button').click(function () {
+                            notify.me({
+                                class: centerPopupClasses,
+                                header: 'Campaign Failed to Create! :(',
+                                subheader: 'You honestly should not be seeing this',
+                                body: 'Something went wrong...which is never a good thing.',
```
Comment (itsmistad) [inline src/DnDTracker.Web/Views/Campaign/Create.cshtml]: Check Slack for recommendations on these line items.

File: src/DnDTracker.Web/Views/Campaign/Create.cshtml [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            </div>
+        </div>
+    </div>
+</div>
+
+@section Scripts
+{
+    <script>
+        $('#create-button').click(function () {
+
```
Comment (itsmistad) [inline src/DnDTracker.Web/Views/Campaign/Create.cshtml:55]: Shouldn't we check that the required fields are filled in before allowing the rest of this? What if I click Create without typing anything?

Comment (TylerCarey98) [reply src/DnDTracker.Web/Views/Campaign/Create.cshtml:55]: How do?

**Review by itsmistad — APPROVED** [review]

**Pull request merged**